core/src/main/java/hudson/model/WorkspaceListener.java

```diff
@@ -3,25 +3,77 @@
 import hudson.ExtensionList;
 import hudson.ExtensionPoint;
 import hudson.FilePath;
+import hudson.Util;
+import hudson.console.ConsoleLogFilter;
+
+import java.io.OutputStream;
 
 public abstract class WorkspaceListener implements ExtensionPoint {
     
     /**
      * Called after a workspace is deleted successfully.
      * @param project
+     * @deprecated as of TODO_BEFORE_MERGE Use {@link #afterDelete(Job)}.
```

Comment:
other comparable workflow refactoring to rely on Job/Run didn't used @Deprecated. 
Any benefit doing so ? javadoc's @deprecated is well supported by all IDEs, isn't it ?
I honestly never understood why this annotation was created :-\